src/tests/test_accesses.py

```diff
@@ -16,12 +15,9 @@ def _patch_session(monkeypatch, mock_table):
     # DB patching
     monkeypatch.setattr(accesses, "accesses", mock_table)
     # Sterilize all DB interactions through CRUD override
-    monkeypatch.setattr(crud, "get", pytest.mock_get)
-    monkeypatch.setattr(crud, "fetch_all", pytest.mock_fetch_all)
-    monkeypatch.setattr(crud, "fetch_one", pytest.mock_fetch_one)
-    monkeypatch.setattr(crud, "post", pytest.mock_post)
-    monkeypatch.setattr(crud, "put", pytest.mock_put)
-    monkeypatch.setattr(crud, "delete", pytest.mock_delete)
+    monkeypatch.setattr(base, "get", pytest.mock_get)
+    monkeypatch.setattr(base, "fetch_all", pytest.mock_fetch_all)
+    monkeypatch.setattr(base, "fetch_one", pytest.mock_fetch_one)
```

Comment:
Since you made the crud __init__, you can keep the previous version (you have to patch the thin imported in the route, and does not import base, but crud)

Reply:
Actually it is not that simple (I think, I am not masting import mechanisms) because it seams that mocking crud.get will keep untouched the original crud.base.get function and if it imported afterward the original is used instead of the mocked version. So far the only way I could mock all crud function was to mock both crud._ and crud.base._ (_ beeing each function). Do you see ?

Reply:
Mmmh, what error did you get with my suggestion?

Because the import for the accesses is over here: https://github.com/pyronear/pyro-api/blob/master/src/app/api/routes/accesses.py#L3

So if we patch this very import, it should be working without any problem :thinking: